### ISSUE TITLE
Copying same object multiple time while writing JSON that contains multiple nested objects

### ISSUE THREAD

**Issue by iamprahlad** [realm/realm-js#3254]
## Goals
Writing JSON that contains multiple nested objects

## Expected Results
```js
{
    id: '12324',
    age: 22,
    name: 'Pankaj',
    phones: [{ model: 'Redmi 5A', price: 60000 }, { model: 'Nokia Lumia X2', price: 5000 }, { model: 'Nokia A1', price: 8000 }],
    car: { model: 'i20', price: 500000, launchDate: new Date(1600028986548) },
    address: {
        addressLocality: "Khatima",
        addressRegion: "Uk",
        postalCode: "262308",
        streetAddress: "Hill View Colony "
    }
}
```

## Actual Results
```js
{
    address: { model: "Redmi 5A", price: 60000, $refId: "RealmObject{0000-0000}" },
    age: 22,
    car: { model: "Redmi 5A", price: 60000, $refId: "RealmObject{0000-0000}" },
    id: "12324",
    name: "Pankaj",
    phones: [
        { model: "Redmi 5A", price: 60000, $refId: "RealmObject{0000-0000}" },
        { model: "Nokia Lumia X2", price: 5000, $refId: "RealmObject{0000-0001}" },
        { model: "Nokia A1", price: 8000, $refId: "RealmObject{0000-0002}" }
    ]
}
```

## Code Sample
```js
const PersonSchema = {
    name: 'Person',
    primaryKey: 'id',
    properties: {
        id: 'string',
        age: 'int',
        name: 'string',
        phones: 'Phone[]',
        car: 'Car',
        address: 'Address'
    }
}

const PhoneSchema = {
    name: 'Phone',
    properties: {
        model: 'string',
        price: 'int'
    }
}

const CarSchema = {
    name: 'Car',
    properties: {
        model: 'string',
        price: 'int',
        launchDate: 'date',
    }
}

const addressSchema = {
    name: 'Address',
    properties: {
        streetAddress: 'string?',
        postalCode: 'string?',
        addressRegion: 'string?',
        addressLocality: 'string?'
    }
}

const dummyPersonData = {
    id: '12324',
    age: 22,
    name: 'Pankaj',
    phones: [{ model: 'Redmi 5A', price: 60000 }, { model: 'Nokia Lumia X2', price: 5000 }, { model: 'Nokia A1', price: 8000 }],
    car: { model: 'i20', price: 500000, launchDate: new Date(1600028986548) },
    address: {
        addressLocality: "Khatima",
        addressRegion: "Uk",
        postalCode: "262308",
        streetAddress: "Hill View Colony "
    }
}


const readTestData = () => {
    Realm.open({ schema: [addressSchema, PersonSchema, CarSchema, PhoneSchema] }).then(realm => {
        try {
            const result = realm.objects('Person')
            console.log('result', result.toJSON())
        } catch (e) {
            console.log('unable to read', e)
        }
        return realm.close();
    }).catch(error => {
        console.log('error ', error)
    })

}

const writeTestData = () => {
    Realm.open({ schema: [addressSchema, PersonSchema, CarSchema, PhoneSchema] }).then(realm => {
        try {
            realm.write(() => {
                realm.create('Person', dummyPersonData)
                console.log('done')
            })
        } catch (e) {
            console.log('unable to write', e)
        }
        return realm.close();
    }).catch(error => {
        console.log('error ', error)
    })
}
```

## Version of Realm and Tooling

- Realm JS SDK Version: ?
- Node or React Native: ?
- Client OS & Version: ?
- Which debugger for React Native: ?/None

**Comment by steffenagger** [comment]
Hello @iamprahlad, thanks for reporting this.
_(I took the liberty to wrap the code-examples in code-blocks for readability, hope that's ok)_

The cause has been found, and a fix will be included in the next release (both v6 & v10-beta).
Just curious, what version of Realm are you currently using?

**Comment by iamprahlad** [comment]
Hi @steffenagger, thanks for help.
I'm using realm version 6.1.1.

**Comment by alexandrethsilva** [comment]
@steffenagger do you know when this fix will be released? I'm currently on 6.1.2 and that release was made 13 days ago. It's a blocking issue for me currently. 🙈

**Comment by steffenagger** [comment]
@alexandrethsilva we're awaiting some [important fixes](https://github.com/realm/realm-core/pull/3957), but the upcoming release should be out one of the following days.
